packages/common/src/ContractInteractor.ts

```diff
@@ -219,7 +219,15 @@ export class ContractInteractor {
     }
     const block = await this.web3.eth.getBlock('latest').catch((e: Error) => { throw new Error(`getBlock('latest') failed: ${e.message}\nCheck your internet/ethereum node connection`) })
     if (block.baseFeePerGas != null) {
-      this.transactionType = TransactionType.TYPE_TWO
+      this.logger.debug('Network supports type two (eip 1559) transactions. Checking rpc node eth_feeHistory method')
+      try {
+        await this.getFeeHistory('0x1', 'latest', [0.5])
+        this.transactionType = TransactionType.TYPE_TWO
+        this.logger.debug('Rpc node supports eth_feeHistory. Initializing to type two transactions')
+      } catch (e: any) {
+        this.logger.debug('eth_feeHistory failed. Aborting.')
```

Comment:
I think the proper error `This is a TYPE-2 network, but node doesn't support eth_feeHistory`